Modules/xSharePoint/DSCResources/MSFT_xSPWebAppPolicy/MSFT_xSPWebAppPolicy.psm1

```diff
@@ -56,63 +81,260 @@ function Set-TargetResource
+
+                    $polbinddiff = Compare-Object -ReferenceObject $policy.PolicyRoleBindings.Name -DifferenceObject $member.PermissionLevel
+                    if ($polbinddiff -ne $null) {
+                        $policy.PolicyRoleBindings.RemoveAll()
+                        foreach ($permissionLevel in $member.PermissionLevel) {
+                            switch ($permissionLevel) {
+                                "Deny All" {
+                                    $policy.PolicyRoleBindings.Add($denyAll)
+                                }
+                                "Deny Write" {
+                                    $policy.PolicyRoleBindings.Add($denyWrite)
+                                }
+                                "Full Control" {
+                                    $policy.PolicyRoleBindings.Add($fullControl)
+                                }
+                                "Full Read" {
+                                    $policy.PolicyRoleBindings.Add($fullRead)
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    # User does not exist. Add user
+                    Write-Verbose -Verbose "Adding $($member.UserName)"
+                    $newPolicy = $wa.Policies.Add($member.UserName, $member.UserName)
+                    foreach ($permissionLevel in $member.PermissionLevel) {
+                        switch ($permissionLevel) {
+                            "Deny All" {
+                                $newPolicy.PolicyRoleBindings.Add($denyAll)
+                            }
+                            "Deny Write" {
+                                $newPolicy.PolicyRoleBindings.Add($denyWrite)
+                            }
+                            "Full Control" {
+                                $newPolicy.PolicyRoleBindings.Add($fullControl)
+                            }
+                            "Full Read" {
+                                $newPolicy.PolicyRoleBindings.Add($fullRead)
+                            }
+                        }
+                    }
+                    if ($member.ActAsSystemAccount) {
+                        $newPolicy.IsSystemUser = $member.ActAsSystemAccount
+                    }
+                }
+                $wa.Update()
             }
+        }
+
+        if ($params.MembersToExclude) {
+            Write-Verbose -Verbose "Processing MembersToExclude parameter"
+
+            $psuAccount = $wa.Properties["portalsuperuseraccount"]
+            $psrAccount = $wa.Properties["portalsuperreaderaccount"]
+
+            Import-Module (Join-Path $ScriptRoot "..\..\Modules\xSharePoint.WebAppPolicy\xSPWebAppPolicy.psm1" -Resolve)
+            
+            foreach ($member in $params.MembersToExclude) {
+                if (($psuAccount -eq $member.UserName) -or ($psrAccount -eq $member.UserName)) {
```

Comment:
True, like the other comment: Realized this when I reviewed your changes to the Cache Account resource. Will change this.